extensions/azure/processors/ListAzureBlobStorage.h

```diff
@@ -0,0 +1,78 @@
+/**
+ * @file ListAzureBlobStorage.h
+ * ListAzureBlobStorage class declaration
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "core/Property.h"
+#include "AzureBlobStorageProcessorBase.h"
+#include "core/logging/LoggerConfiguration.h"
+
+class ListAzureBlobStorageTestsFixture;
+
+namespace org::apache::nifi::minifi::azure::processors {
+
+class ListAzureBlobStorage final : public AzureBlobStorageProcessorBase {
+ public:
+  SMART_ENUM(EntityTracking,
+    (NONE, "none"),
+    (TIMESTAMPS, "timestamps")
+  )
```

Comment:
In my opinion it would be better to integrate this lib in a separate PR if it proves to be better than our current implementation. That way we can replace all the other occurrences of SMART_ENUM as well and have only a single enum library in our codebase instead of a mixing the two.